Form/Type/SecurityRolesType.php

```diff
@@ -41,33 +37,6 @@ public function __construct(EditableRolesBuilder $rolesBuilder)
     /**
      * {@inheritdoc}
      */
-    public function buildForm(FormBuilderInterface $formBuilder, array $options)
-    {
-        /*
-         * The form shows only roles that the current user can edit for the targeted user. Now we still need to persist
-         * all other roles. It is not possible to alter those values inside an event listener as the selected
-         * key will be validated. So we use a Transformer to alter the value and an listener to catch the original values
-         *
-         * The transformer will then append non editable roles to the user ...
-         */
-        $transformer = new RestoreRolesTransformer($this->rolesBuilder);
```

Comment:
Have you checked if all hidden roles (the user have no permissions to) will stay?

Reply:
Ping?